### PR TITLE
ET-3619 Budget module

### PR DESCRIPTION
**Pull request by jorge-cr-13**
## What 🕵️ 🔍

- Creating a module for budget and the terragrunt files. 

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/jira/software/projects/ET/boards/51?selectedIssue=ET-3619)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: aws/b2b_personalization/modules/budgets/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+```hcl
+module "budgets" {
+  source = "../../modules/budgets"
+
+  budget_limit       = "100"
+  threshold_value    = 80
+  notification_email = ["test@test.com"]
+  
+  budget_tags = {
+    Environment = "Test"
```
Comment (Robert-Steiner) [inline aws/b2b_personalization/modules/budgets/README.md:16]: that is cool! 👍

File: aws/b2b_personalization/global/budgets/terragrunt.hcl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+terraform {
```
Comment (Robert-Steiner) [inline aws/b2b_personalization/global/budgets/terragrunt.hcl:1]: we can actually move this to `_env` and `dev`

and then write
```hcl
  budget_tags = merge(
    local.env_vars.locals.tags,
    {
      User      = "shared"
    }
  )
```

File: aws/b2b_personalization/modules/budgets/variables.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+variable "threshold_value" {
+  description = "Threshold in percentage for notifying"
+  type        = number
+  default     = 80
+}
+
+variable "notification_email" {
+  description = "List of emails to send the budget notifications"
+  type        = list(string)
+  default     = ["devops@xayn.com"]
```
Comment (Robert-Steiner) [inline aws/b2b_personalization/modules/budgets/variables.tf:21]: we could move the email address to our sops file. wdyt?

Comment (jorge-cr-13) [reply aws/b2b_personalization/modules/budgets/variables.tf]: Yes, sounds like a very good idea.

**Review by Robert-Steiner — APPROVED** [review]

**Pull request merged**